sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -585,16 +585,10 @@ class Dataset[T] private[sql](
    * @group basic
    * @since 3.0.0
    */
-  def explain(mode: String): Unit = sparkSession.withActive {
-    // Because temporary views are resolved during analysis when we create a Dataset, and
-    // `ExplainCommand` analyzes input query plan and resolves temporary views again. Using
-    // `ExplainCommand` here will probably output different query plans, compared to the results
-    // of evaluation of the Dataset. So just output QueryExecution's query plans here.
-
+  def explain(mode: String): Unit =
```

Comment:
Hmmm .. can't you just hijack stdout and save it into a string?

Reply:
I can and do and see others also do, so was thinking looks like missing feature